apps/fabric-website/src/components/IconGrid/IconGrid.tsx

```diff
@@ -36,12 +36,21 @@ export class IconGrid extends React.Component<IIconGridProps, IIconGridState> {
         <ul className={ styles.grid }>
           { icons
             .filter(icon => icon.name.toLowerCase().indexOf(searchQuery.toLowerCase()) !== -1)
-            .map((icon, iconIndex) => (
-              <li key={ iconIndex } aria-label={ icon.name + ' icon' }>
-                <i className={ `ms-Icon ms-Icon--${icon.name}` } title={ `${icon.name}` } aria-hidden='true' />
-                <span>{ icon.name }</span>
-              </li>
-            ))
+            .map((icon, iconIndex) => {
+              let iconJsxElement = <i ref={ (`${icon.name}`) } className={ `ms-Icon ms-Icon--${icon.name}` } title={ `${icon.name}` } aria-hidden='true' />;
+              let iconRefElement = this.refs[icon.name] as HTMLElement;
+
+              if (iconRefElement && iconRefElement.offsetWidth > 80) {
+                iconJsxElement = <i ref={ (`${icon.name}`) } className={ `ms-Icon ms-Icon--${icon.name} hoverIcon` } title={ `${icon.name}` } aria-hidden='true' />;
```

Comment:
Searched high and low for a better way to add the 'hoverIcon' class name, but with the immutable nature of the react element's class, this was the best I could come up with. Might be able to try something with set state, too. 

Alternatively, we could skip the class altogether and just give every icon a background color on hover.

Reply:
Can you use the `css()` utility to add that class conditionally? Otherwise I think you could do something like this:

```
const hoverIconClass: string = iconRefElement && iconRefElement.offsetWidth > 80 ? 'hoverIcon' : '';
let iconJsxElement = <i ref={ (`${icon.name}`) } className={ `ms-Icon ms-Icon--${icon.name} ${hoverIconClass}` } title={ `${icon.name}` } aria-hidden='true' />;
```

But I might be misunderstanding something here. It's not a lot of repeated code but I agree it would be best to find a way to clean it up. :)

Reply:
The problem I was having with conditional class naming is that I needed to create the JSX element before being able to check the width. Any ideas for getting around that? Would love to clean it up.

Reply:
Ahh, I get it now. I think this would have to be moved to `componentDidMount()`, which will be called after the icons are rendered into the DOM. That would give access to the rendered width. While that should work, it will immediately re-render the entire icon grid, which could be a performance issue. This is an interesting problem... might be fine to just leave what you have since it's working. :)

Reply:
Great points. I'll leave this up here a little longer to see if anyone else has any ideas.